pkg/core/resource_graph.go

```diff
@@ -26,14 +26,30 @@ func (rg *ResourceGraph) AddResource(resource Resource) {
 	}
 }
 
-// AddDependency2 deliberately renamed from AddDependency in the short term to catch when dependencies were
-// being added in the inverse order. If either `source` or `dest` don't exist in the graph, they are added.
-func (rg *ResourceGraph) AddDependency2(source Resource, dest Resource) {
-	for _, res := range []Resource{source, dest} {
+// Adds a dependency such that `deployedSecond` has to be deployed before `deployedFirst`. This makes the left-to-right
+// association consistent with our visualizer, and with the Go struct graph.
+//
+// For example, if you have a Lambda and its execution role, then:
+//
+//	╭────────────────╮   ╭────────────────╮
+//	│ LambdaFunction ├──➤│    IamRole     │
+//	│ deployedSecond │   │ deployedFirst  │
+//	╰────────────────╯   ╰────────────────╯
+//
+// And you would invoke it as:
```

Comment:
```suggestion
// And you would construct it as:
```

nit: "invoke" doesn't seem quite the right word to me

Reply:
ah, good catch. ("invoke" was from when I just had the method call, without the struct). What do you think about "use" (since the method call is still in the snippet, so "construct" doesn't quite work either)?